app/models/notification.rb

```diff
@@ -0,0 +1,34 @@
+class Notification < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+  belongs_to :user
+  belongs_to :course
+
+  scope :center_popup, -> { where(type: 'CenterPopup') }
+  scope :right_side_popup, -> { where(type: 'RightSidePopup') }
+  scope :email, -> { where(type: 'EmailNotification') }
+
+  def self.types
+    %w(CenterPopup RightSidePopup EmailNotification)
+  end
+
+  def self.create(user, course, options = {})
```

Comment:
As for create, that is my fault. I did not mean to overriding that function. I will change its name.
And also, I will add a description in google doc